TheEthicalHackerCup/Assets/Scripts/SecurityConceptProgress.cs

```diff
@@ -25,13 +26,15 @@ public SecurityConceptProgress(int maxDefenseUpgradeLevel = 5)
     public int GetAttackMinigamesAttempted() { return this._attackMinigamesAttempted; }
     public int GetHeat() { return this._heat; }
 
+    public int GetAttackMinigameAttemptsRequiredToUpgrade() { return Math.Min(this._maxDefenseUpgradeLevel, this._currentDefenseUpgradeLevel + 1); }
```

Comment:
Why this equation?
Double checking, it looks like what you're saying is the first level, we need to do 1 attack minigame, the second level 2 and then finally once we make it past the max upgrade level, we just stick at the upgrade level.

If we already hit the max, why would we still be trying to upgrade the attacks?

Reply:
If the player is at max upgrade level, they would not be able to upgrade again, but they would be shown the attempts required to get to their final upgrade (current implementation). 

The attacksRequiredForNextUpgrade would be for the denominator for the attacksAttempted/attacksRequiredForNextUpgrade fraction shown in the defense upgrade menu

If the player attempts A LOT of attacks, it would show something like 10AttemptsDone/3AttemptedRequired to upgrade.
In this case the player would be able to repeatedly upgrade until they reach the max, if the oppKnowlege gets distributed fast enough to allow for that

I guess that raises the question of "should the player be allowed to attempt atk minigames more than the minimum attempts required to fully upgrade". A point for group discussion I suppose.